#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.26963264991124586 0.1297722209745497 -0.9541789165377593 2.280560678914944
  position 4.216298602420491 6.14722088135461 5.094605313836252
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation -0.09 0 0.0300019
  rotation 0 1 0 5.30716e-06
  floorSize 4 4
}
DEF Obs1 Solid {
  translation 1.82357 1.49541 0.142956
  rotation -0.0025754902978973823 0.002671230308971273 0.9999931156654838 -1.5881053071795863
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 1 0.05 0.5
      }
    }
  ]
}
DEF Obs1 Solid {
  translation -0.192899 -0.590293 0.132619
  rotation -0.0025754902978973823 0.002671230308971273 0.9999931156654838 -1.5881053071795863
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 1 0.05 0.5
      }
    }
  ]
  name "solid(5)"
}
DEF Obs1 Solid {
  translation -0.984034 0.50358 0.128187
  rotation -0.0025754902978973823 0.002671230308971273 0.9999931156654838 -1.5881053071795863
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 1 0.05 0.5
      }
    }
  ]
  name "solid(4)"
}
Robot {
  translation -1.67847 -1.78965 0.0823736
  rotation -0.0003814181485216626 -0.0038827415119134395 -0.999992389390313 -1.5364853071795865
  children [
    DEF gps GPS {
      translation -0.05 -0.02 0.03
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Cylinder {
            height 0.02
            radius 0.02
          }
        }
      ]
      boundingObject Shape {
        appearance PBRAppearance {
        }
        geometry Cylinder {
          height 0.02
          radius 0.02
        }
      }
      physics Physics {
      }
    }
    DEF lidar Lidar {
      translation 0.07 -0.02 0.04
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Cylinder {
            height 0.025
            radius 0.025
          }
        }
      ]
      boundingObject Shape {
        appearance PBRAppearance {
        }
        geometry Cylinder {
          height 0.025
          radius 0.025
        }
      }
      physics Physics {
      }
      horizontalResolution 128
      fieldOfView 3.14
      verticalFieldOfView 0.1
      numberOfLayers 1
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.2 0.1 0.05
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        position 8330.682749169071
        axis 0 1 0
        anchor 0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.04973415452170613 0.06043224674532349 -0.0008681479169474751
        rotation 0.8542902265985025 -0.3670974908825199 0.3680049468771274 1.7503610170350365
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.305882 0.898039 0.25098
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        position 5994.624329950352
        axis 0 1 0
        anchor 0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation 0.049999594167756005 -0.06000008469764562 2.183249105056611e-06
        rotation -0.9481315443993817 -0.2247734507711365 0.22477426530472455 4.659148406503984
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        position 8330.648217447058
        axis 0 1 0
        anchor -0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation -0.050004115097078375 0.060000084994313085 -2.150657114429141e-06
        rotation -0.8404581071865204 0.3831638405631713 -0.3831652924644039 4.539445739519625
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        position 5994.6237586462075
        axis 0 1 0
        anchor -0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.050000654251758384 -0.05999991499841628 1.9981940367764346e-06
        rotation -0.948253408417907 -0.22451626440827555 0.22451708273531307 4.659276742487037
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF DS_RIGHT DistanceSensor {
      translation 0.1 -0.03 0
      rotation 0 0 1 -0.3
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
    }
    DEF DS_LEFT DistanceSensor {
      translation 0.1 0.03 0
      rotation 0 0 1 0.3
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
    }
  ]
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "controlador"
  window "<none>"
}
Solid {
  translation 1.24 -1.22 0
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.9 0.05 0.6
      }
    }
  ]
  name "solid(1)"
}
Solid {
  translation -0.05 1.03 0.09
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 2 0.05 0.6
      }
    }
  ]
  name "solid(2)"
}
Solid {
  translation -0.98 -0.93 0.09
  rotation 0 0 1 -1.5707953071795862
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 2 0.05 0.6
      }
    }
  ]
  name "solid(3)"
}
